src/WingetCreateCore/Common/PackageParser.cs

```diff
@@ -255,16 +333,37 @@ public static bool UpdateInstallerNodes(InstallerManifest installerManifest, IEn
             }
         }
 
-        private static bool ParsePackageAndGenerateInstallerNodes(string path, string url, List<Installer> installers, Manifests manifests)
+        private static bool ParsePackageAndGenerateInstallerNodes(
+            string path,
+            string url,
+            List<Installer> installers,
+            Manifests manifests,
+            ref List<DetectedArch> detectedArchOfInstallers)
         {
             var installer = new Installer();
             installer.InstallerUrl = url;
             installer.InstallerSha256 = GetFileHash(path);
             installer.Architecture = GetMachineType(path)?.ToString().ToEnumOrDefault<InstallerArchitecture>() ?? InstallerArchitecture.Neutral;
 
-            return ParseExeInstallerType(path, installer, installers) ||
+            bool parseResult = ParseExeInstallerType(path, installer, installers) ||
                 ParseMsix(path, installer, manifests, installers) ||
                 ParseMsi(path, installer, manifests, installers);
+
+            var archGuess = GetArchFromUrl(installer.InstallerUrl);
+
+            if (!archGuess.HasValue)
```

Comment:
You can shorten this to var archGuess = GetArchFromUrl(..) ?? installer.InstallerUrl;

Reply:
That said, why set this at all? It's not a guess anymore if you overwrite it?

Reply:
I think this could all be re-written as:

            var archGuess = GetArchFromUrl(installer.InstallerUrl);
            detectedArchOfInstallers.Add(new DetectedArch(installer.InstallerUrl, archGuess.Value, installer.Architecture));

            if (archGuess.HasValue)
            {
                installer.Architecture = archGuess.Value;
            }

Reply:
Because GetArchFromUrl returns a nullable type, archGuess.Value will throw an invalidoperationexception if there is no value.

Reply:
Ah, so just archGuess then, and DetectedArch.UrlArch should be nullable

Reply:
Changed to suggested with DetectedArch.UrlArch as nullable